### PR TITLE
LPS 156207 LPS 154443 LPS 157320

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
- LPS-156207 Remove feature flag and usages in code
- LPS-156207 Remove usages in tests
- LPS-154443 Add tests for the depot DM remote staging
- LPS-156207 Remove usages in Java tests
- LPS-157320 Return file version on the cases where the URL is a frindly url tp show the preview

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
see https://github.com/liferay-lima/liferay-portal/pull/3200#issuecomment-1171318875

File: modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/servlet/filter/DLVideoEmbedFilter.java [hunk 1]
```diff
@@ -184,10 +207,89 @@ private String _getFileVersionId(HttpServletRequest httpServletRequest) {
...
+	}
+
+	private User _getUser(HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		HttpSession httpSession = httpServletRequest.getSession();
+
+		if (PortalSessionThreadLocal.getHttpSession() == null) {
+			PortalSessionThreadLocal.setHttpSession(httpSession);
+		}
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/servlet/filter/DLVideoEmbedFilter.java:232]: Why do we need this?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/servlet/filter/DLVideoEmbedFilter.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/servlet/filter/DLVideoEmbedFilter.java:232]: Reviewing the code, I think that is not needed.
It's from the original class where I copy the method to obtain the user.

Deleted and pushed

Thanks!!!

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by liuxingye** [comment]
Reviewing...

**Comment by liuxingye** [comment]
[LocalFile.DepotCollections#CanBePublishedViaRemoteStaging](https://testray.liferay.com/home/-/testray/case_results/1893647122) is not related to this PR.

**Comment by liuxingye** [comment]
Hi @AliciaGarciaGarcia, I found the following behavior when fixing the Poshi tests. Could you take a look it when you get a chance? Thanks.

1. Add an asset library > Upload an image to it like Document_2.jpg
2. Connect it to the Liferay site
3. Go to Liferay site > Web Content > Add a basic article and select the image from asset library
4. Preview the Web Content
5. Inspect the image part > Now the element from HTML is document_2
6. Edit the Document_2.jpg to Document_3.jpg (Update both the filename and title to Document_3.jpg)
7. Preview the WC again
8. Now the image has changed to Document_3 but the element in HTML is still document_2

**Comment by AliciaGarciaGarcia** [comment]
@liuxingye 
I think that it's the expected behavior if you did not deleted the old friendlyURL of _Document_2_. It will redirect to the document, that it the case you mentioned is the Document_3.jpg
The content of the WC never changes if you change the Document

**Comment by liuxingye** [comment]
@AliciaGarciaGarcia  thanks for looking into the issues. I've resend a new pull with Poshi changes https://github.com/liferay-lima/liferay-portal/pull/3208

cc @austinchiang

**Pull request closed without merging**